tutorials/arvo/cli-tutorial.md

```diff
@@ -0,0 +1,420 @@
+`[%delete path]`. This is not required though, and you could use something like
+`[chat-room=@t =action]`.
+
+`card` is either an ordinary Gall agent `card` or a `%shoe` `card`, which takes
+the shape `[%shoe sole-ids=(list @ta) effect=shoe-effect]`. A `%shoe` `card` is
+sent to all `sole`s listed in `sole-ids`, imaking them run the `sole-effect`
+specified by `effect` (i.e. printing some text).  Here we can
+reference `card:shoe` because of `/+  shoe` at the beginning of the app.
+
+```hoon
+=|  state-0
+=*  state  -
+::
+```
+Add the bunt value of `state-0` to the head of the subject, then give it the
+macro `state`. The `-` here is a lark expression referring to the head of the
+subject. This allows us to use `state` to refer to the state elsewhere in the
+code no matter what version we're using, while also getting direct access to the
+contents of `state` (if it had any).
+
+```hoon
+%+  verb  |
+%-  agent:dbug
+^-  agent:gall
+%-  (agent:shoe command)
+^-  (shoe:shoe command)
+```
+
+The casts here are just reminders of what is being produced. So let's focus on
+what the `%` runes are doing, from bottom to top. We call `(agent:shoe command)`
+on what follows (i.e. the rest of the app),
+producing a standard Gall agent core. Then we call wrap the Gall agent core with
+`agent:dbug`, endowing it with additional arms useful for debugging, and then
+wrap again with `verb`.
+
+
+```hoon
+|_  =bowl:gall
++*  this  .
+    def   ~(. (default-agent this %|) bowl)
+    des   ~(. (default:shoe this command) bowl)
+```
+This is boilerplate Gall agent core code. We set `this` to be a macro for the
+subject, which is the Gall agent core itself. We set `def` and `des` to be
+macros for initialized `default-agent` and `default:shoe` doors respectively.
+
+Next we implement all of the arms required for a `shoe` agent. Starting with
+the standard Gall arms:
+
+```hoon
+++  on-init   on-init:def
+++  on-save   !>(state)
+++  on-load
+  |=  old=vase
+  ^-  (quip card _this)
+  [~ this]
+::
+++  on-poke   on-poke:def
+++  on-watch  on-watch:def
+++  on-leave  on-leave:def
+++  on-peek   on-peek:def
+++  on-agent  on-agent:def
+++  on-arvo   on-arvo:def
+++  on-fail   on-fail:def
+```
+These are minimalist Gall app arm implementations using the default behavior found in `def`.
+
+Here begins the implementation of the additional arms required by the
+`(shoe:shoe command)` interface.
+```hoon
+++  command-parser
+  |=  sole-id=@ta
+  ^+  |~(nail *(like [? command]))
+  (cold [& ~] (jest 'demo'))
+```
+`+command-parser` is of central importance - it is what is used to parse user
+input and transform it into `command`s for the app to execute. Writing a proper
+command parser requires understanding of the Hoon parsing functions found in the
+standard library. How to do so may be found in the parsing documentation (coming
+soon). For now, it is sufficient to know that this arm matches the text "demo" and
+produces a `[? command]`-shaped noun in response.
```

Comment:
```suggestion
produces a `[? command]`-shaped noun in response. Note how the `&` signifies that the command will be run as soon as it has been entered, without waiting for the user to press return.
```

Reply:
Oh this explains why I remember having to press Enter. I was using an older version of the code at the time that did not have this flag (which I wasn't totally sure what it did either until now).